src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/FootnoteServiceTests.cs

```diff
@@ -32,33 +34,742 @@ namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf(indicator.Id),
+                SetOf<Guid>());
+
+            result.AssertRight();
+            Assert.Single(result.Right.Indicators);
+            Assert.Equal(indicator.Id, result.Right.Indicators.Single().IndicatorId);
+        }
+
+        [Fact]
+        public async Task CreateFootnote_WithFiltersAndIndicatorsAndSubjectsWhichAreAllLinkedToRelease_CreatesFootnoteWithCorrectLinks()
+        {
+            Release release = _fixture.DefaultStatsRelease().Generate();
+            FilterItem filterItem = _fixture.DefaultFilterItem().Generate();
+            FilterGroup filterGroup = _fixture.DefaultFilterGroup().WithFilterItems(new List<FilterItem>() { filterItem }).Generate();
+            Filter filter = _fixture.DefaultFilter().WithFilterGroups(new List<FilterGroup>() { filterGroup }).Generate();
+            Indicator indicator = _fixture.DefaultIndicator().Generate();
+            IndicatorGroup indicatorGroup = _fixture.DefaultIndicatorGroup().WithIndicators(new List<Indicator>() { indicator }).Generate();
+            Subject subject1 = _fixture.DefaultSubject().WithFilters(new List<Filter>() { filter }).WithIndicatorGroups(new List<IndicatorGroup>() { indicatorGroup }).Generate();
+            Subject subject2 = _fixture.DefaultSubject().Generate();
+            ReleaseSubject releaseSubject1 = _fixture.DefaultReleaseSubject().WithRelease(release).WithSubject(subject1).Generate();
+            ReleaseSubject releaseSubject2 = _fixture.DefaultReleaseSubject().WithRelease(release).WithSubject(subject2).Generate();
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await SeedDatabase(contextId, release, subjects: ListOf(subject1, subject2), releaseSubjects: ListOf(releaseSubject1, releaseSubject2));
+
+            Either<ActionResult, Footnote> result = await CreateFootnoteWithConfiguration(
+                release.Id,
+                contextId,
+                string.Empty,
+                SetOf(filter.Id),
+                SetOf(filterGroup.Id),
+                SetOf(filterItem.Id),
+                SetOf(indicator.Id),
+                SetOf(subject2.Id));
+
+            result.AssertRight();
+            Assert.Single(result.Right.Subjects);
+            Assert.Equal(subject2.Id, result.Right.Subjects.Single().SubjectId);
+            Assert.Single(result.Right.Filters);
+            Assert.Equal(filter.Id, result.Right.Filters.Single().FilterId);
+            Assert.Single(result.Right.FilterGroups);
+            Assert.Equal(filterGroup.Id, result.Right.FilterGroups.Single().FilterGroupId);
+            Assert.Single(result.Right.FilterItems);
+            Assert.Equal(filterItem.Id, result.Right.FilterItems.Single().FilterItemId);
+            Assert.Single(result.Right.Indicators);
+            Assert.Equal(indicator.Id, result.Right.Indicators.Single().IndicatorId);
+        }
```

Comment:
I think this is now very similar to `CreateFootnote`. Just wondering if we can combine them so we don't need both?

Reply:
I've deleted the old one and added the content check into the new one :)